### PR TITLE
[SYCLomatic] Support migration of cudaOccupancyMaxPotentialBlockSize

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

File: clang/runtime/dpct-rt/include/util.hpp.inc [hunk 1]
```diff
@@ -914,6 +914,30 @@ inline int calculate_max_active_wg_per_xecore(int *num_wg, int wg_size,
...
+/// \param [in] sg_size Sub-group size.
+/// \param [in] used_barrier Whether barrier is used.
+/// \param [in] used_large_grf Whether large General Register File is used.
+/// \return Returns 0.
+inline int calculate_max_potential_wg(int *num_wg, int *wg_size,
+                                      int max_ws_size_for_device_code,
+                                      int slm_size = 0, int sg_size = 32,
+                                      bool used_barrier = false,
+                                      bool used_large_grf = false) {
+  size_t max_wg_size = dev.get_info<sycl::info::device::max_work_group_size>();
```
Comment (tangjj11) [inline clang/runtime/dpct-rt/include/util.hpp.inc]: `dev` is undefined.

**Review by tangjj11 — CHANGES_REQUESTED** [review]

File: clang/runtime/dpct-rt/include/util.hpp.inc (hunk 1 above)
Comment (zhiweij1) [reply clang/runtime/dpct-rt/include/util.hpp.inc]: fixed

File: clang/lib/DPCT/APINames.inc [hunk 2]
```diff
@@ -358,7 +358,7 @@ ENTRY(cudaGraphInstantiateWithParams, cudaGraphInstantiateWithParams, false, NO_
 
 
 // C++ high level API functions of runtime API
-ENTRY(cudaOccupancyMaxPotentialBlockSize, cudaOccupancyMaxPotentialBlockSize, false, NO_FLAG, P0, "no direct mapping yet", "NA")
+ENTRY(cudaOccupancyMaxPotentialBlockSize, cudaOccupancyMaxPotentialBlockSize, true, NO_FLAG, P0, "no direct mapping yet", "NA")
```
Comment (zhimingwang36) [inline clang/lib/DPCT/APINames.inc]: need file the warnning message. DPCT1111

Comment (zhiweij1) [reply clang/lib/DPCT/APINames.inc]: done

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**